onyo/lib/onyo.py

```diff
@@ -833,6 +833,50 @@ def generate_faux_serials(self, length: int = 6, num: int = 1) -> set[str]:
 
         return faux_serials
 
+    def valid_name(self, asset: Path) -> bool:
+        """
+        Verify that an asset name fits into the asset name scheme of onyo:
+        <type>_<make>_<model>.<serial>
+        Where the fields type, make, and model do not allow the characters '.'
+        and '_' to be part of the field, and no field is allowed to be empty.
+
+        Returns True for valid asset names, and False if invalid.
+        """
+        # verify that a dot exists
+        if asset.name.find(".") == -1:
+            log.error("Asset names must have a '.'")
+            return False
+
+        # split fields
+        try:
+            # reverse order, because re reads from left to right and splits,
+            # and this way '.' and '_' are in serial field
+            [serial, model, make, type] = re.findall('(.*)\.(.*)_(.*)_(.*)', asset.name[::-1])[0]
+        except (ValueError, IndexError):
+            log.error(f"'{asset.name}' must be like '<type>_<make>_<model>.<serial>'")
+            return False
+
+        # check against special characters in fields 'type', 'make', 'model'
+        if '_' in type or '.' in type or '_' in make or '.' in make or '_' in model or '.' in model:
+            log.error(f"'{asset.name}' must be like '<type>_<make>_<model>.<serial>'")
+            return False
+
+        # none of the fields is allowed to be empty
+        if not type:
```

Comment:
Since the change already drops the information which field is empty (which is fine), I change it not to `any()`, but to `if not all([type, make, model, serial]):`